ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -345,6 +362,68 @@ public static BlobProperties buildBlobProperties(Map<String, Object> args)
   }
 
   /**
+   * Build a {@link ByteRange} given the arguments associated with a request.
+   * @param args the arguments associated with the request.
+   * @return The {@link ByteRange} parsed from the arguments, or {@code null}, if no range argument was found
+   * @throws RestServiceException if the range could not be parsed from the header value, or it was otherwise invalid.
+   */
+  public static ByteRange buildByteRange(Map<String, Object> args)
+      throws RestServiceException {
+    String rangeValue = getHeader(args, Headers.RANGE, false);
+    if (rangeValue == null) {
+      return null;
+    }
+    if (!rangeValue.startsWith(BYTE_RANGE_PREFIX)) {
+      throw new RestServiceException("Invalid byte range syntax; does not start with 'bytes='",
+          RestServiceErrorCode.RangeNotSatisfiable);
+    }
+    try {
+      int hyphenIndex = rangeValue.indexOf('-', BYTE_RANGE_PREFIX.length());
+      String startOffsetStr = rangeValue.substring(BYTE_RANGE_PREFIX.length(), hyphenIndex);
+      String endOffsetStr = rangeValue.substring(hyphenIndex + 1);
+      if (startOffsetStr.isEmpty()) {
+        return ByteRange.fromLastNBytes(Long.parseLong(endOffsetStr));
+      } else if (endOffsetStr.isEmpty()) {
+        return ByteRange.fromStartOffset(Long.parseLong(startOffsetStr));
+      } else {
+        return ByteRange.fromOffsetRange(Long.parseLong(startOffsetStr), Long.parseLong(endOffsetStr));
+      }
+    } catch (Exception e) {
```

Comment:
This can catch any exception that occurred while parsing/forming the byte range. This means that a range could not be created, and a `RestServiceException` will be bubbled up.